INSTALL.html

```diff
@@ -1,368 +1,336 @@
 </ul>
-<br>
-FYI, <code>Metamath.zip</code> is just a directory. "Install" it by
-just unzipping it to the "C" drive, which creates directory "<code
- style="font-weight: bold;">c:\metamath</code>". And that's it for
-that...unless you want to add "<code>c:\metamath</code>" to the Windows
-environment "<code>path</code>" (see II below).<br>
-<br>
-The <code>mmj2.zip</code> download provides certain handy .bat files,
-including these that build html pages:<br>
+
+<p>
+An alternative way to install Java would be to install an
+<a href="https://www.oracle.com/java/technologies/javase-downloads.html"
+>Oracle Java release</a>; make absolutely <i>certain</i> that you
+comply with Oracle's licenses if you do this
+(the free one, in particular, has many restrictions on how you
+can legally use it).
+
+<p>
+<h3>(Recommended) Install git and a good text editor</h3>
+<p>
+The "git" program is a very widely used version control system
+that supports global collaborative development of software and
+software-like artifacts.
+You can get a Windows version from the
+<a href="https://git-scm.com/downloads">Git SCM downloads</a>.
+<p>
+You will almost certainly need a good text editor that can handle
+large files, and preferably one that can handle POSIX standard line endings
+(LF instead of Windows' CRLF).
+The programs <tt>Notepad</tt> and <tt>Wordpad</tt> are <b>not</b>
+good text editors.
+There are many available; if you have no idea, you might try
+<a href="https://notepad-plus-plus.org/">Notepad++</a>, which is small,
+simple, and gets the job done.
+Other options include <a href="https://atom.io/">Atom</a>,
+<a href="https://www.vim.org/">vim</a>
+<a href="https://code.visualstudio.com/">Microsoft Visual Studio Code</a>.
+
+<p>
+<h3>Install mmj2</h3>
+<p>
+Install mmj2; we recommend putting it in "C:\mmj2" on Windows.
+Click on the "start" button, type "command", press ENTER, and then type:
+
+<pre>
+cd c:\
+git clone https://github.com/digama0/mmj2.git
+cd mmj2
+</pre>
+
+
+<p>
+<h3>(Optional) Compile mmj2</h3>
+<p>
+You do not have to compile mmj2, because we provide a
+precompiled jar file.
+If you want to recompile it, here's what you do.
+<p>
+First, if you directly downloaded the mmj2 git repository, as recommended, run
+the following from the top-level "mmj2" directory:
+<pre>
+git submodule init
+git submodule update
+</pre>
+<p>
+Now to actually compile it in Windows, presuming it's in its conventional
+C:\mmj2 location, run:
+
+<pre>
+C:\mmj2\compile\windows\CompMMJ.bat
+</pre>
+
+<p>
+This creates the <code>c:\mmj2\mmj2jar\mmj2jar.jar</code>
+jar file, in addition to creating the various mmj2 class files.<br>
+
+<p>
+You should also compile the Javadoc for mmj2.
+Use the Command Prompt window to execute the following command:<br>
+<code>C:\mmj2\doc\windows\BuildDoc.bat</code>
+
+<p>
+<h3>(Recommended) Install the Metamath database set.mm</h3>
+<p>
+Put the database set.mm in its recommended place, which is
+C:\set.mm for Windows.
```

Comment:
I'm not sure I like these new recommended default locations. This is a highly nonstandard place to put things in Windows, and it is also inconvenient for people who have set.mm from a git repo (which we should be encouraging). How about using `C:\set.mm\set.mm` instead?

Really, I think we should not have any "default" location. Although it's fine to have these paths appear in the repo version of the file, these install instructions should point out every place where the user needs to change a path to make it fit where they have actually put things. Saying that this weird place is "recommended" without saying what the consequences of doing otherwise is just intimidation.

Reply:
If there is to be a default location for set.mm, then it would be nice, imho, to have a common "metamath databases" folder, i.e. have files
~/mmdb/set.mm
~/mmdb/iset.mm
or, if another level of nesting is needed (why?):
~/mmdb/set.mm/set.mm
~/mmdb/iset.mm/iset.mm

Reply:
I certainly have a `metamath` folder containing subfolders for all metamath related projects, including `metamath-exe/`, `mmj2/`, `set.mm/` and all the other verifiers I have had cause to use.

Reply:
In my mmj2 startup script I use paths like `../../set.mm/set.mm` to refer to `metamath/set.mm/set.mm` from `metamath/mmj2/mmj2jar/mmj2.jar`. This survives migration to other folders when I switch computers better than `C:\metamath` does. (But I nevertheless have a hardlink in this location because it makes certain programs easier to use, including mmj2. However, setting up hardlinks in windows is a real pain, needing to be done via `cmd`.)

Reply:
I have a folder organization similar to you, with
~/Documents/metamath/mm182/ [containing the metamath program, version 0.182]
~/Documents/metamath/mmfiles/ [containing databases]
~/Documents/metamath/mmj2/
and I use relative paths like `../../mm182/metamath`.
So as long as it's not too hard to maintain this organization (with or without setting defaults), I'm fine.

Reply:
When I say "put in "C:\set.mm' I mean the *directory* "c:\set.mm" - filename would be c:\set.mm\set.mm. The problem is that as currently written the mmj2 scripts want to know ahead-of-time fixed locations, yet people don't use the same conventions for file locations. Hmm. I'll post separately on this. mmj2 just doesn't work like most programs, making it more painful to do the simple things.

Reply:
Legitimate issues. I'll post some things here:

https://github.com/digama0/mmj2/issues/39

Reply:
I'm working to solve this by creating a rewritten mmj2 front-end that can be easily *told* on the command line what database to use (including the path), and if you don't tell it, have some reasonable smarts to try to find it.

Reply:
Okay, I think I've solved this by writing a new "mmj2" script (and its equivalent mmj2.bat) that searches many likely locations, lets you easily say "-d DATABASE_PATH", and remembers the last one used.